src/main/java/hudson/plugins/release/pipeline/ReleaseStepExecution.java

```diff
@@ -0,0 +1,108 @@
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+
+import com.google.inject.Inject;
+
+import hudson.AbortException;
+import hudson.console.ModelHyperlinkNote;
+import hudson.model.Action;
+import hudson.model.BuildableItemWithBuildWrappers;
+import hudson.model.Cause;
+import hudson.model.CauseAction;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.ParameterDefinition;
+import hudson.model.ParameterValue;
+import hudson.model.Queue;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.release.ReleaseWrapper;
+import hudson.plugins.release.SafeParametersAction;
+import jenkins.model.Jenkins;
+import jenkins.model.ParameterizedJobMixIn;
+
+/**
+ * @author Alexey Merezhin
+ * @since 2.7
+ */
+public class ReleaseStepExecution extends StepExecution {
+    private static final Logger LOGGER = Logger.getLogger(ReleaseStepExecution.class.getName());
+
+    transient ReleaseStep step;
+
+    public ReleaseStepExecution(@Nonnull StepContext context, @Nonnull ReleaseStep step) {
+        super(context);
+        this.step = step;
+    }
+
+    private List<ParameterValue> updateParametersWithDefaults(ParameterizedJobMixIn.ParameterizedJob project,
+            List<ParameterValue> parameters) throws AbortException {
+
+        if (project instanceof BuildableItemWithBuildWrappers) {
+            ReleaseWrapper wrapper = ((BuildableItemWithBuildWrappers) project).getBuildWrappersList()
+                                                                               .get(ReleaseWrapper.class);
+            if (wrapper != null) {
+                for (ParameterDefinition pd : wrapper.getParameterDefinitions()) {
+                    boolean parameterExists = false;
+                    for (ParameterValue pv : parameters) {
+                        if (pv.getName().equals(pd.getName())) {
+                            parameterExists = true;
+                            break;
+                        }
+                    }
+                    if (!parameterExists) {
+                        parameters.add(pd.getDefaultParameterValue());
+                    }
+                }
+            } else {
+                throw new AbortException("Job doesn't have release plugin configuration");
+            }
+        }
+        return parameters;
+    }
+
+    @Override
+    public boolean start() throws Exception {
+        if (step.getJob() == null) {
+            throw new AbortException("Job name is not defined.");
+        }
+
+        final ParameterizedJobMixIn.ParameterizedJob project = Jenkins.getActiveInstance().
+                getItemByFullName(step.getJob(), ParameterizedJobMixIn.ParameterizedJob.class);;
+        if (project == null) {
+            throw new AbortException("No parametrized job named " + step.getJob() + " found");
+        }
+        getContext().get(TaskListener.class).getLogger()
```

Comment:
Maybe retrieval of TaskListener needs a null check. Just to make FindBugs happy once it is enabled